File: 김남형/Week2/BOJ_10845_큐.py
import sys

n = int(sys.stdin.readline())
arr = []

for i in range(n):
  com = sys.stdin.readline().split()
  
  if com[0] == 'push':
    arr.append(com[1])
  elif com[0] == 'size':
    print(len(arr))
  elif com[0] == 'pop':
    if len(arr) > 0:
      print(arr.pop(0))
    else:
      print(-1)
  elif com[0] == 'empty':
    if len(arr) > 0:
      print(0)
    else:
      print(1)
  elif com[0] == 'front':
    if len(arr) > 0:
      print(arr[0])
    else:
      print(-1)
  elif com[0] == 'back':
    if len(arr) > 0:
      print(arr[-1])
    else:
      print(-1)

File: 김한비/Week2/BOJ_10845_큐.cpp
#include <iostream>
#include <queue>
#include <string>

using namespace std;

int main() {
	int n;
	queue<int> q;

	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	cin >> n;
	for (int i = 0; i < n; i++) {
		string cmd;
		cin >> cmd;

		if (cmd == "pop") {
			if (q.size() == 0)
				cout << -1 << '\n';
			else {
				cout << q.front() << '\n';
				q.pop();
			}
		}
		else if (cmd == "size")
			cout << q.size() << '\n';
		else if (cmd == "empty") {
			if (q.size() == 0)
				cout << 1 << '\n';
			else
				cout << 0 << '\n';
		}
		else if (cmd == "front") {
			if (q.size() == 0)
				cout << -1 << '\n';
			else
				cout << q.front() << '\n';
		}
		else if (cmd == "back") {
			if (q.size() == 0)
				cout << -1 << '\n';
			else
				cout << q.back() << '\n';
		}
		else if (cmd == "push") {
			int num;
			cin >> num;
			q.push(num);
		}
	}

	return 0;
}